app/routes/questions/add.tsx

```diff
@@ -0,0 +1,74 @@
+import {useState} from 'react'
+import type { ActionArgs } from '@remix-run/cloudflare'
+import { useActionData, Form, useSearchParams, useTransition, useSubmit } from '@remix-run/react'
+import { redirect } from '@remix-run/cloudflare'
+import { addQuestion, loadAllQuestions } from '~/server-utils/stampy'
+
+
+export const action = async ({ request }: ActionArgs) => {
+    const formData = await request.formData();
+    let title = formData.get('title')
+    const state = formData.get('stateString');
+    const redirectTo = '/' + (state ? '?state=' + state : '')
+    const relatedQuestions = formData.getAll('relatedQuestion').map(question => ({ title: question }));
+
+    // Make sure that the question was provided
+    if (!title) return redirect(redirectTo);
+
+    // Check whether the question is simply a prefix of an existant question,
+    // and if so ignore it
+    const allQuestions = await loadAllQuestions(request);
+    const isPrefix = allQuestions.data.some(
+        question => question.toLowerCase().startsWith(title.toLowerCase())
+    );
+    if (isPrefix) return redirect(redirectTo);
+
+    // Make sure the question is formatted as a question
+    if (!title.endsWith('?')) title = title + '?'
+    title = title[0].toUpperCase() + title.substring(1);
+
+    const result = await addQuestion(title, relatedQuestions);
+    console.log('Added question "' + title + '", response:', result);
+
+    return redirect(redirectTo);
+};
+
+export const AddQuestion = ({ title, relatedQuestions, onQuestionAdded }) => {
+    const actionData = useActionData<typeof action>();
+    const [remixSearchParams] = useSearchParams();
+    const [stateString] = useState(() => remixSearchParams.get('state'));
+
+    const transition = useTransition();
+    const submit = useSubmit();
+
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        submit(e.currentTarget, { replace: true });
+        onQuestionAdded(title);
+    }
+
+    return (
+        <Form method="post"
+              action="/questions/add"
+              className="result-item none-of-the-above"
+              title="Request a new question"
+              onSubmit={handleSubmit}
+        >
+            <input type="hidden" name="title" value={title}/>
+            <input type="hidden" name="stateString" value={stateString}/>
+            {relatedQuestions.map(title => (
+                <input type="hidden"
+                       name="relatedQuestion"
+                       key={title}
+                       value={title}
+                />
+            ))}
+           {transition.state === "submitting"
+               ? "Adding question..."
```

Comment:
this isn't really needed, as submitting this form also results in its destruction

Reply:
we shouldn't destroy this while loading, can you please keep it visible (don't blur / change detection of blur to keep the search form visible while interacting with the links, please)

Reply:
Won't that mean that after clicking on this, the user won't see any action until the question gets added to Coda? Or do you mean that adding a question shouldn't close the suggested answers box?

Reply:
I removed the transition stuff